core/notification/src/main/java/edu/stanford/spezi/core/notification/setting/NotificationSettingScreen.kt

```diff
@@ -0,0 +1,200 @@
+import edu.stanford.spezi.core.design.component.CenteredBoxContent
+import edu.stanford.spezi.core.design.theme.Colors.primary
+import edu.stanford.spezi.core.design.theme.Spacings
+import edu.stanford.spezi.core.design.theme.SpeziTheme
+import edu.stanford.spezi.core.design.theme.TextStyles
+import edu.stanford.spezi.core.design.theme.ThemePreviews
+import edu.stanford.spezi.core.notification.R
+
+@Composable
+fun NotificationSettingScreen() {
+    val viewModel = hiltViewModel<NotificationSettingViewModel>()
+    val uiState by viewModel.uiState.collectAsState()
+    NotificationSettingScreen(
+        onAction = viewModel::onAction,
+        uiState = uiState,
+    )
+}
+
+@Composable
+internal fun NotificationSettingScreen(
+    onAction: (NotificationSettingViewModel.Action) -> Unit,
+    uiState: NotificationSettingViewModel.UiState,
+) {
+    Scaffold(topBar = {
+        AppTopAppBar(title = {
+            Text(
+                text = stringResource(R.string.notification_settings),
+            )
+        }, navigationIcon = {
+            IconButton(onClick = {
+                onAction(NotificationSettingViewModel.Action.Back)
+            }) {
+                Icon(
+                    Icons.AutoMirrored.Filled.ArrowBack,
+                    contentDescription = stringResource(R.string.back)
+                )
+            }
+        })
+    }, content = { paddingValues ->
+        Column(
+            modifier = Modifier
+                .padding(paddingValues)
+                .padding(horizontal = Spacings.medium)
+        ) {
+            when (uiState) {
+                is NotificationSettingViewModel.UiState.Error -> {
+                    CenteredBoxContent {
+                        Text(
+                            text = uiState.message,
+                            style = TextStyles.headlineMedium,
+                            textAlign = TextAlign.Center,
+                        )
+                    }
+                }
+
+                NotificationSettingViewModel.UiState.Loading -> {
+                    CenteredBoxContent {
+                        CircularProgressIndicator(color = primary)
+                    }
+                }
+
+                is NotificationSettingViewModel.UiState.NotificationSettingsLoaded -> NotificationOptions(
+                    notificationSettings = uiState.notificationSettings,
+                    onAction = onAction,
+                    pendingActions = uiState.notificationSettings.pendingActions,
+                )
+            }
+        }
+    })
+}
+
+@Composable
+private fun NotificationOptions(
+    notificationSettings: NotificationSettings,
+    onAction: (NotificationSettingViewModel.Action) -> Unit,
+    pendingActions: PendingActions<NotificationType>,
+) {
+    LazyColumn(modifier = Modifier.padding(vertical = Spacings.medium)) {
+        notificationSettings.groupBySection().forEach { (section, settings) ->
```

Comment:
Can we do this above using remember? I would move the extension function here in this file

```kotlin
@Composable
private fun NotificationSettings.groupBySection(): Map<NotificationType.Section, List<Pair<NotificationType, Boolean>>> {
    return remember(key1 = this) {
        entries.groupBy(
            keySelector = { it.key.section },
            valueTransform = { it.toPair() }
        )
    }
}
```